### PR TITLE
Use tagger to send container/orchestrator metadata tags

### PR DESCRIPTION
**Pull request by sunhay**
We'll now be sourcing all container & orchestrator metadata from a single source: [`/util/tagger` in datadog-agent](https://github.com/DataDog/datadog-agent/blob/master/pkg/tagger).

The tagger currently collects and includes docker/kubernetes metadata + labels, and ecs metadata.

This results in lot of duplicated metadata in so we'll be omitting:
- ` KubeMetadataPayload`
- top-level fields: `name`, `id`, `image`; in `model.Container`

**Config updates**
- The following configuration variables move from **datadog-process.ini** to **datadog.yaml**:
`DD_KUBERNETES_KUBELET_HOST` -> `kubernetes_kubelet_host`
`DD_KUBERNETES_KUBELET_HTTP_PORT` -> `kubernetes_http_kubelet_port`
`DD_KUBERNETES_KUBELET_HTTPS_PORT` -> `kubernetes_https_kubelet_port`
`DD_COLLECT_KUBERNETES_METADATA` -> Not available

**Example containers:**
```
 "containers": [
    {
      "type": "Docker",
      "cpuLimit": 100,
      "state": 3,
      "created": 1513116841,
      "netRcvdPs": 2,
      "netSentPs": 1,
      "netRcvdBps": 1018,
      "netSentBps": 66,
      "memRss": 10182656,
      "memCache": 3903488,
      "started": 1513116841,
      "tags": [
        "kube_deployment:kube-dns",
        "kube_namespace:kube-system",
        "kube_container_name:sidecar",
        "pod_name:kube-dns-86f6f55dd5-sdskb",
        "kube_replica_set:kube-dns-86f6f55dd5",
        "docker_image:gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.5",
        "image_name:gcr.io/google_containers/k8s-dns-sidecar-amd64",
        "image_tag:1.14.5",
        "container_name:k8s_sidecar_kube-dns-86f6f55dd5-sdskb_kube-system_7c77b2a5-dea5-11e7-bdb9-080027e5e101_9",
        "container_id:2b6aff691c57a8087d83dd9cc0cd20a2194f8343d9ff667cfd193a0a02bf668c"
      ]
    },
    {
      "type": "Docker",
      "cpuLimit": 100,
      "state": 3,
      "created": 1513116840,
      "netRcvdPs": 2,
      "netSentPs": 1,
      "netRcvdBps": 1018,
      "netSentBps": 66,
      "memRss": 4923392,
      "memCache": 2244608,
      "started": 1513116841,
      "tags": [
        "kube_deployment:kube-dns",
        "kube_namespace:kube-system",
        "kube_container_name:dnsmasq",
        "pod_name:kube-dns-86f6f55dd5-sdskb",
        "kube_replica_set:kube-dns-86f6f55dd5",
        "image_name:gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64",
        "image_tag:1.14.5",
        "docker_image:gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.5",
        "container_name:k8s_dnsmasq_kube-dns-86f6f55dd5-sdskb_kube-system_7c77b2a5-dea5-11e7-bdb9-080027e5e101_9",
        "container_id:75a5a103c0f4581afaec6bee2f368f34a8e1ea0bb763a10db685467a963a0e35"
      ]
    }
  ],
```

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -8,9 +8,7 @@ imports:
   subpackages:
   - gogen
 - name: github.com/DataDog/datadog-agent
-  version: a1a945b5c4cbf64ee2762a71c25d6229c57a4840
-  repo: git@github.com:DataDog/datadog-agent.git
-  vcs: git
+  version: 92c410ac3af9d73d237fe076d77d1d365b9abe5c
```
Comment (sunhay) [inline glide.lock]: https://github.com/DataDog/datadog-agent/commit/92c410ac3af9d73d237fe076d77d1d365b9abe5c

This just rebases https://github.com/DataDog/datadog-agent/pull/847 off of master so I could test with the latest kubelet tagger changes.

**Comment by sunhay** [comment]
We could also probably remove `ECSMetadataPayload` + our ecs package, but the ecs tagger collector hasn't yet been updated to support Fargate. Thoughts on just removing it and making sure it gets updated as part of https://github.com/DataDog/datadog-agent/pull/847?

File: checks/container.go [hunk 2]
```diff
@@ -57,7 +57,6 @@ func (c *ContainerCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.Me
 
 	// Fetch orchestrator metadata once per check.
 	ecsMeta := ecs.GetMetadata()
```
Comment (conorbranagan) [inline checks/container.go]: can ecs be dropped too?

**Review by conorbranagan — CHANGES_REQUESTED** [review]
This looks like a good first pass.

I'm worried about backwards-compatibility with Agent 5. While we are pushing to Agent 6 now, we will still have versions of our agent fall in v5.

Can we create a fake yaml config with sane defaults when there is no yaml available to keep things working as they are now?

File: checks/container.go (hunk 2 above)
Comment (sunhay) [reply checks/container.go]: It doesn't look like the ecs tagger collector supports Fargate at the moment, so I was hesitant to remove it as it'd be a slight regression. I'm not 100% sure if it'll be added as part of DataDog/datadog-agent#847.

Comment (conorbranagan) [reply checks/container.go]: OK that's fair, thanks. We can leave in until that's ready
cc @hkaj

**Comment by sunhay** [comment]
I think I may be a little confused re: agent 5 compatibility and exactly how the process agent code depends on it. 

The current defaults removed by these changes the same defaults as the `datadog.yaml` (https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config.go#L135). Maybe I'm missing something?

**Review by conorbranagan — APPROVED** [review]

**Pull request merged**